src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -473,31 +495,31 @@ CHIP_ERROR Spake2p_P256_SHA256_HKDF_HMAC::KDF(const uint8_t * ikm, const size_t
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR EcdsaRawSignatureToAsn1(size_t fe_length_bytes, const uint8_t * raw_sig, size_t raw_sig_length, uint8_t * out_asn1_sig,
-                                   size_t out_asn1_sig_length, size_t & out_asn1_sig_actual_length)
+CHIP_ERROR EcdsaRawSignatureToAsn1(size_t fe_length_bytes, const ByteSpan & raw_sig, MutableByteSpan out_asn1_sig,
+                                   size_t & out_asn1_sig_actual_length)
```

Comment:
We shouldn't need that last arg; just resize `out_asn1_sig` to have the right length using `reduce_size`, so it represents the actual output, and take it by non-const lvalue reference.

Reply:
As per previous comment: I had to learn of a non-standard `Span` mutator: `reduce_size`, on top of needing to possibly use what I consider to be an immutable *wrapper* around a buffer (a span) to actually be mutated.

BTW I found it confusing to have "MutableByteSpan" that seems to imply the span is mutable (which it is, but oddly). I initially expected ConstByteSpan and ByteSpan rather than ByteSpan and MutableByteSpan, respectively.

Reply:
Yes, the "MutableByteSpan" naming is not great and we should fix it...